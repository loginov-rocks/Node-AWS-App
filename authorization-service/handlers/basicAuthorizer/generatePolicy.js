/**
 * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html#api-gateway-lambda-authorizer-lambda-function-create
 */
export default (principalId, isAllowed, resource) => ({
  principalId,
  policyDocument: {
    Version: '2012-10-17',
    Statement: [
      {
        Action: 'execute-api:Invoke',
        Effect: isAllowed ? 'Allow' : 'Deny',
        Resource: resource,
      },
    ],
  },
});
