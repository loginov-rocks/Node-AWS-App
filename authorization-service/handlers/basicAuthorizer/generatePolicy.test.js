import generatePolicy from './generatePolicy';

it('generates valid policy', () => {
  expect(generatePolicy('principal', true, 'arn')).toMatchSnapshot();
});
