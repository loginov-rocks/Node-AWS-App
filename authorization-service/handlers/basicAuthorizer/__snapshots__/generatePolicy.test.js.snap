// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates valid policy 1`] = `
Object {
  "policyDocument": Object {
    "Statement": Array [
      Object {
        "Action": "execute-api:Invoke",
        "Effect": "Allow",
        "Resource": "arn",
      },
    ],
    "Version": "2012-10-17",
  },
  "principalId": "principal",
}
`;
